saleor/graphql/order/types.py

```diff
@@ -184,14 +185,20 @@ class Meta:
     @staticmethod
     @traced_resolver
     def resolve_user(root: models.OrderEvent, info):
-        user = info.context.user
-        if (
-            user == root.user
-            or user.has_perm(AccountPermissions.MANAGE_USERS)
-            or user.has_perm(AccountPermissions.MANAGE_STAFF)
-        ):
-            return root.user
-        raise PermissionDenied()
+        def _resolve_user(event_user):
+            user = info.context.user
+            if (
+                user == event_user
+                or user.has_perm(AccountPermissions.MANAGE_USERS)
+                or user.has_perm(AccountPermissions.MANAGE_STAFF)
+            ):
+                return event_user
+            return None
+
+        if not root.user_id:
+            return None
```

Comment:
I know it was like that before, but shouldn't we consider also an app requestor here?

Reply:
good point, will check other loaders as well.

Reply:
I'm not sure about this permission, because if I'm staff user responsible only for order. I can't checkout who fulfill or process other actions on this order. 
IMO all staff users able to see order events should be able to see at least the email of staff who generate that event.
PS. Maybe we should do it in separate PR for all events. By implementing resolver like, `event_performer_email`

Reply:
@pwgryglak 
What do you think?

Reply:
Users should know who created the event - that is the purpose of having a history in first place.